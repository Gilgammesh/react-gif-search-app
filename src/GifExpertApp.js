import React, { useState } from "react";
import PropTypes from "prop-types";
import CategoryAdd from "./components/CategoryAdd";
import GifGrid from "./components/GifGrid";

const GifExpertApp = ({ defaultCategorias }) => {
  const [categorias, setCategorias] = useState(defaultCategorias);

  const categList = categorias.map((ele, index) => {
    return <GifGrid key={index} categoria={ele} />;
  });

  return (
    <>
      <h2>GifExpertApp</h2>
      <CategoryAdd setCategorias={setCategorias} />
      <br />
      <hr />
      {categList}
    </>
  );
};

GifExpertApp.propTypes = {
  defaultCategorias: PropTypes.array,
};

GifExpertApp.defaultProps = {
  defaultCategorias: ["Naruto"],
};

export default GifExpertApp;
