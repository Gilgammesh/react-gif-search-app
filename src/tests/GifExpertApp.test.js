import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import GifExpertApp from "../GifExpertApp";

describe("Pruebas unitarias en el componente GifExpertApp.js", () => {
  const categorias = ["Samurai X", "Saitama", "Naruto"];
  /************************************************************************************/
  test("se debe mostrar el componente <GifExpertApp /> correctamente", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });
  /************************************************************************************/
  test("se debe mostrar una lista de categorias", () => {
    const wrapper = shallow(<GifExpertApp defaultCategorias={categorias} />);
    expect(wrapper).toMatchSnapshot();
    const grids = wrapper.find("GifGrid");
    expect(grids.length).toBe(categorias.length);
  });
});
