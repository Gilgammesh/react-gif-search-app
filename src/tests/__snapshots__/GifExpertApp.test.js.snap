// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas unitarias en el componente GifExpertApp.js se debe mostrar el componente <GifExpertApp /> correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <CategoryAdd
    setCategorias={[Function]}
  />
  <br />
  <hr />
  <GifGrid
    categoria="Naruto"
    key="0"
  />
</Fragment>
`;

exports[`Pruebas unitarias en el componente GifExpertApp.js se debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <CategoryAdd
    setCategorias={[Function]}
  />
  <br />
  <hr />
  <GifGrid
    categoria="Samurai X"
    key="0"
  />
  <GifGrid
    categoria="Saitama"
    key="1"
  />
  <GifGrid
    categoria="Naruto"
    key="2"
  />
</Fragment>
`;
